### PR TITLE
Remove text about extension service.

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
Occam's Razor. It's unnecessary complication that makes this text harder to read. I don't agree. The job of the specification is to document external behavior.

I don't agree that it makes future specs more difficult. They can just say "allocate a context ID". It's an internal detail of the implementation that there is a context ID allocation service.

**Comment by DavidSchinazi** [comment]
How about the following? It removes the concept but keeps the requirements we need.

```
Context IDs are 62-bit integers (0 to 2<sup>62</sup>-1). Context IDs are encoded
as variable-length integers, see {{Section 16 of QUIC}}. The context ID value of
0 is reserved for UDP payloads, while non-zero values are dynamically allocated:
non-zero even-numbered context IDs are client-allocated, and odd-numbered
context IDs are server-allocated. Implementations are responsible for only allocating
each context ID once, though they need not allocate them in order.  Once allocated, any
context ID can be used by both client and server - only allocation carries
separate namespaces to avoid requiring synchronization. Additionally, note that
the context ID namespace is tied to a given HTTP request: it is possible for a
context ID with the same numeric value to be simultaneously assigned different
semantics in distinct requests, potentially with different semantics.
```

**Comment by afrind** [comment]
I prefer removing the allocation service as a concept and specifying the requirements.

I did find this a bit confusing in David's proposed text

>  - only allocation carries separate namespaces to avoid requiring synchronization

I think the point is clear already and the rationale is not required.  Maybe remove this clause?

**Comment by ekr** [comment]
This text still has the problem that it first says you can allocate each context ID once and then later it says actually you can allocate it once per namespace. Why not say this upfront as in my PR?

File: draft-ietf-masque-connect-udp.md [hunk 1]
```diff
@@ -285,22 +285,21 @@ separate from UDP payloads. In order to accomplish this, all HTTP Datagrams
...
+server-allocated. The context ID namespace is tied to a given HTTP
+request: it is possible for a context ID with the same numeric value
+to be simultaneously assigned different semantics in distinct
+requests, potentially with different semantics.  Context IDs must not
+be re-allocated within a given HTTP namespace but MAY be allocated in
+any order. Once allocated, any context ID can be used by both client
+and server - only allocation carries separate namespaces to avoid
+requiring synchronization.
+[Optional: Implementations are responsible for arbitrating requests
+for context IDs from multiple extensions.]
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp.md:302]: I'd suggest leaving this Optional bit out for now
```suggestion
```

**Review by DavidSchinazi — APPROVED** [review]
My apologies @ekr I think I had missed the added sentence in the middle when looking at this the first time. This LGTM modulo removal of the optional sentence at the end.

**Pull request merged**